bundles/org.openhab.binding.deconz/src/test/java/org/openhab/binding/deconz/DeconzTest.java

```diff
@@ -92,6 +94,7 @@ public void discoveryTest() throws IOException {
         Mockito.doAnswer(answer -> CompletableFuture.completedFuture(Optional.of(bridgeFullState))).when(bridgeHandler)
                 .getBridgeFullState();
         ThingDiscoveryService discoveryService = new ThingDiscoveryService();
+        discoveryService.modified(Map.of(DiscoveryService.CONFIG_PROPERTY_BACKGROUND_DISCOVERY, false));
```

Comment:
I'm not sure about that, because that is not the default setting. Maybe removing `discoveryService.startScan()` and relying on the background discovery would be an approach that is closer to a production system.

Reply:
Ok. I can change it.

.. but on a side note, it seems to me that the Junit test is not really a test of the binding code, but rather of the OH core..

Reply:
Why? It ensures that the discovery code creates a `DiscoveryResult` for every device in the input JSON. I would say that this is binding code.

Reply:
.. but in any case, removing the `startScan()` does also fails the Junit test (see below) .. so evidently we cannot _'rely on the background discovery'_ -- of the three options i) foreground scan with no background, ii) background with no foreground, or iii) both background and foreground, it seems to me that i) (my solution) is the most controlled test and therefore one most likely to pass properly. 

```
[ERROR] Failures:
[ERROR]   DeconzTest.discoveryTest:100
Wanted but not invoked:
discoveryListener.thingDiscovered(
    <any>,
    <any>
);
-> at org.openhab.binding.deconz.DeconzTest.discoveryTest(DeconzTest.java:100)

However, there was exactly 1 interaction with this mock:
discoveryListener.removeOlderResults(
    org.openhab.binding.deconz.internal.discovery.ThingDiscoveryService@329a1f8d,
    0L,
    [deconz:colorcontrol, deconz:dimmablelight, deconz:powersensor, deconz:consumptionsensor, deconz:moisturesensor, deconz:waterleakagesensor, deconz:onofflight, deconz:switch, deconz:vibrationsensor, deconz:openclosesensor, deconz:humiditysensor, deconz:colortemperaturelight, deconz:extendedcolorlight, deconz:windowcovering, deconz:airqualitysensor, deconz:lightsensor, deconz:batterysensor, deconz:pressuresensor, deconz:presencesensor, deconz:alarmsensor, deconz:colorlight, deconz:doorlock, deconz:firesensor, deconz:temperaturesensor, deconz:carbonmonoxidesensor, deconz:daylightsensor, deconz:warningdevice, deconz:thermostat],
    null
);
-> at org.openhab.core.config.discovery.AbstractDiscoveryService.removeOlderResults(AbstractDiscoveryService.java:330)
```